Jropp/jason-ropp-js-iobp
# Onboarding Project: Validate email

## What It Does

### Email Validation

Adds email validation and warning messages to the user component.

Previously the user card allowed any input into the email field. It now checks for proper format and disables functionality when the format is not correct.

I added code to `formWatcher()` and `updateWarningMessages()` in user-component.html that both sets messages and disable/enables save appropriately.

```
if (emailInProgress) {
          this.emailFormatMessage = email.checkEmailInput(this.user.email) ? '' : emailWarning;
        }
```

```
this.disableSave = (this.editOpen && !this.user.id) || this.fieldsFilled() || !phone.checkPhoneInput(this.user.phone) || !email.checkEmailInput(this.user.email);
```

Fixes parts of #55 

### Utils.js

To simplify `user-component.html`'s growing code I created a `utils.js` file that contains functions related to phone and email checking. 

## How To Test

Enter an improper email and try to save. Fix the email and then save. When not properly entered, the save button should also be disabled. 

## Checklist

Under penalty of public shaming, I avow that I:


Browsers tested:

## Dependencies

- [ ] #56
